https://github.com/emscripten-ports/SDL2/issues/66
Remove when emscripten-ports/SDL2 is tagged version_16 and grabbed by emcc

commit 33f3c0d50b4aa12b98611d621caeb2dab68f837a
Author: Charlie Birks <charlie@daftgames.net>
Date:   Sat Dec 8 14:06:35 2018 +0000

    Emscripten: use a fake size for external sizing check
    
    The check would fail if the canvas happened to be the correct size
    already. (#66, mentioned in #58)

diff --git a/src/video/emscripten/SDL_emscriptenvideo.c b/src/video/emscripten/SDL_emscriptenvideo.c
index 897d47096..5206c0393 100644
--- a/src/video/emscripten/SDL_emscriptenvideo.c
+++ b/src/video/emscripten/SDL_emscriptenvideo.c
@@ -206,8 +206,8 @@ Emscripten_CreateWindow(_THIS, SDL_Window * window)
     scaled_w = SDL_floor(window->w * wdata->pixel_ratio);
     scaled_h = SDL_floor(window->h * wdata->pixel_ratio);
 
-    emscripten_set_canvas_size(scaled_w, scaled_h);
-
+    /* set a fake size to check if there is any CSS sizing the canvas */
+    emscripten_set_canvas_size(1, 1);
     emscripten_get_element_css_size(NULL, &css_w, &css_h);
 
     wdata->external_size = SDL_floor(css_w) != scaled_w || SDL_floor(css_h) != scaled_h;
@@ -217,10 +217,11 @@ Emscripten_CreateWindow(_THIS, SDL_Window * window)
         scaled_w = css_w * wdata->pixel_ratio;
         scaled_h = css_h * wdata->pixel_ratio;
 
-        emscripten_set_canvas_size(scaled_w, scaled_h);
         SDL_SendWindowEvent(window, SDL_WINDOWEVENT_RESIZED, css_w, css_h);
     }
 
+    emscripten_set_canvas_size(scaled_w, scaled_h);
+
     /* if the size is not being controlled by css, we need to scale down for hidpi */
     if (!wdata->external_size) {
         if (wdata->pixel_ratio != 1.0f) {
